test/e2e/federated-service.go

```diff
@@ -50,6 +50,9 @@ const (
 	FederatedServicePod  = "federated-service-test-pod"
 
 	DefaultFederationName = "federation"
+
+	// We use this to decide how long to wait for our DNS probes to succeed.
+	DNSTTL = 180 * time.Second
```

Comment:
Ideally you should rather make the following public, and reference that directly, otherwise that and this could get out of sync, and break the tests in the future.

https://github.com/kubernetes/kubernetes/blob/master/federation/pkg/federation-controller/service/dns.go#L29